compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -287,6 +289,37 @@ pub unsafe fn create_module<'ll>(
         );
     }
 
+    // Pass on the control-flow protection flags to LLVM (equivalent to `-fcf-protection` in Clang).
+    match sess.opts.cg.cf_protection {
+        CFProtection::None => {}
+        CFProtection::Branch => llvm::LLVMRustAddModuleFlag(
+            llmod,
+            llvm::LLVMModFlagBehavior::Override,
```

Comment:
I'm not entirely sure if `Override` is the best option here, but since clang does the same :shrug:.